pkg/components/contour/component_test.go

```diff
@@ -22,41 +22,72 @@ import (
 	"github.com/kinvolk/lokomotive/pkg/components/util"
 )
 
-func testRenderManifest(t *testing.T, configHCL string) {
-	body, diagnostics := util.GetComponentBody(configHCL, name)
-	if diagnostics != nil {
-		t.Fatalf("Error getting component body: %v", diagnostics)
-	}
-
-	component := newComponent()
-	diagnostics = component.LoadConfig(body, &hcl.EvalContext{})
-	if diagnostics.HasErrors() {
-		t.Fatalf("Valid config should not return error, got: %s", diagnostics)
-	}
-
-	m, err := component.RenderManifests()
-	if err != nil {
-		t.Fatalf("Rendering manifests with valid config should succeed, got: %s", err)
-	}
-	if len(m) <= 0 {
-		t.Fatalf("Rendered manifests shouldn't be empty")
-	}
-}
-
-func TestRenderManifestWithIngressHosts(t *testing.T) {
-	configHCL := `
+//nolint:funlen
+func TestRenderManifest(t *testing.T) {
+	tests := []struct {
+		desc    string
+		hcl     string
+		wantErr bool
+	}{
+		{
+			desc: "With ingress hosts",
+			hcl: `
 component "contour" {
   ingress_hosts = ["test.domain.com"]
 }
-`
-	testRenderManifest(t, configHCL)
-}
-
-func TestRenderManifestWithServiceMonitor(t *testing.T) {
-	configHCL := `
+			`,
+		},
+		{
+			desc: "With service monitor",
+			hcl: `
 component "contour" {
   service_monitor = true
 }
-`
-	testRenderManifest(t, configHCL)
+			`,
+		},
+		{
+			desc: "With service type",
+			hcl: `
+component "contour" {
+  service_type = "NodePort"
+}
+			`,
+		},
+		{
+			desc: "Wrong service type",
+			hcl: `
+component "contour" {
+  service_type = "stuff"
+}
+			`,
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range tests {
+		b, d := util.GetComponentBody(tc.hcl, name)
+		if d != nil {
+			t.Fatalf("Error getting component body: %v", d)
+		}
+
+		c := newComponent()
+		d = c.LoadConfig(b, &hcl.EvalContext{})
+
+		if !tc.wantErr && d.HasErrors() {
+			t.Fatalf("Valid config should not return error, got: %s", d)
```

Comment:
Why `Fatalf` instead of `Errorf`? There is no point if one error is found in one test case, to run the other test cases too? (With error, it will fail but continue to run the others)

IMHO, it is useful to run the other testcases too. In that case, just `Errorf()` seems better, or if you want to abort execution of the current test too, `Errorf()` and `continue` (to continue with the next item in the loop). What do you think?

Reply:
@rata what I think should happen is that each case should run in `t.Run()` scope, so then use of `t.Fatal` is legitimate.

Reply:
I don't think it makes sense to use `t.Parallel()`, it has some side-effects and this probably takes some ms to run.

In any case, this is nit-picking IMHO. I'm fine with this being as it is too.

Reply:
Good point @rata. This matters because I've switched to table-driven tests, which means all test cases for Contour are executed in a single function. To clarify, `t.Fatal` only stops any additional tests for the Contour component. Subsequent test **functions** continue executing as expected.

Reply:
Changed to `t.Errorf`.